### ISSUE TITLE
Request: Ability to disable ipv6

### ISSUE THREAD

**Issue by poflynn** [sous-chefs/postfix#125]
### Cookbook version

Latest
### Chef-client version

Latest
### Platform Details

CentOS 7
### Scenario:

Unless I'm missing something (which is possible and apologies if so) but it seems like I can't disable ipv6. Note from the man page below ipv6 is always used first but our SMTP server  is running ipv4 only so I'd like to force it to only use ipv4 using the inet_protocols directive as below. 

Thanks.

```
postconf(5)

 man page:

When IPv6 support is enabled via the inet_protocols parameter,  Post-
fix will do DNS type AAAA record lookups.

When  both IPv4 and IPv6 support are enabled, the Postfix SMTP client
will attempt to connect via IPv6 before attempting to use IPv4.

Examples:

inet_protocols = ipv4
inet_protocols = all (DEFAULT)
inet_protocols = ipv6
inet_protocols = ipv4, ipv6
```

**Comment by randyrue** [comment]
I'm also trying to make this happen. If I'm reading the thing right, should I be able to put the line:

node['postfix']['main']['inet_protocols'] = 'ipv4'

in my role's cookbook/attributes/default.rb file?

**Comment by tas50** [comment]
I’m adding the Type: Jump In GitHub label to this issue. This is a great issue for someone to get their feet wet with and we’d love a PR to resolves the issue.

**Comment by mvangoor** [comment]
We ran into this issue aswell now and the problem was introduced in v5.4.1 where the postmap has been changed to run 'immediately' without first updating the config files.

We now hard pinned on v5.4.0....

**Comment by ramereth** [comment]
@mvangoor can you please provide some example code which replicates this issue? I want to make sure I get the fix correct.

**Comment by mvangoor** [comment]
@ramereth I opened #171 for that.

Basically install a fresh centos 8 vm (but could be any distro I guess) that has no ipv6 enabled (sysctl ipv6 disabled). Then run this cookbook and you will see it will try to postmap on a not valid main.conf (ipv6 is expected to be enabled) and it will fail.

I checked the cookbook and could find no reason why the writing of the master.conf and main.conf was after the postmap bit, but I might be missing something
